modules/openapi-generator/src/main/java/org/openapitools/codegen/InlineModelResolver.java

```diff
@@ -407,11 +407,18 @@ private void fixStringModel(Schema m) {
         }
     }
 
+    /**
+     * Generates a unique model name. Non-alphanumeric characters will be replaced
+     * with underscores
+     *
+     * @param title String title field in the schema if present
+     * @param key String model name
+     */
     private String resolveModelName(String title, String key) {
         if (title == null) {
-            return uniqueName(key);
+            return uniqueName(key).replaceAll("[^A-Za-z0-9]", "_");
```

Comment:
are `$` or `_` not allowed to be in the model name?

Reply:
I think `$` would mean something else in some scripting languages

Reply:
@wing328 how about `_`?

Reply:
Very good question. The spec doesn't restrict the use of symbols (e..g $, _, .) in model naming.

This is auto-generated model name based on the parent model name and the property name. The issue occurs when the parent model name contains `.`.

Since the model name is auto-generated if `title` attribute is not present, this fix will simply get rid of non-alphanumeric characters to make our life easier.

If the user wants complete control of the model naming, they can still use `title` (https://github.com/OpenAPITools/openapi-generator/pull/3498/commits/35e49b9ef66556d34edba357d8a897caf7932ec5)

Reply:
Why does it matter if this name has a dot in it? Should we instead change `uniqueName` so it doesn't generate a name with a dot in it?

Reply:
> Why does it matter if this name has a dot in it? 

It's causing a bug later that the model (created on the fly) cannot be looked up. Again this is auto-generated "schema" name, which will be used to determine the model name using `toModelName`.

This change has no impact on the schema name in the spec provided by the user.